### ISSUE TITLE
Support Plasma 6 (Port to KF6)

### ISSUE THREAD

**Issue by DarkXero-dev** [MarianArlt/kde-plasmoid-chiliclock#26]
Please port this widget to KDE Framework 6. The guidelines are [here](https://develop.kde.org/docs/plasma/widget/porting_kf6/).

**Comment by Svenum** [comment]
I would love to see this too

**Comment by DarkXero-dev** [comment]
Seeing that last commit was 4 years ago I wouldn't hold my breath xD

**Comment by Svenum** [comment]
I know...

**Comment by Svenum** [comment]
I am trying to port it on my own to Plasma6. But I have trouble testing it because of unstable-nixos. Would you try it @DarkXero-dev?
```bash
git clone https://github.com/Svenum/kde-plasmoid-betterinlineclock.git
cd kde-plasmoid-betterinlineclock
git checkout plasma6
plasmoidviewer -a ./org.kde.plasma.betterinlineclock
```

**Comment by DarkXero-dev** [comment]
I would except am currently sick as a dog and don't have access to PC.. Will do in the morning if I feel better.

**Comment by DarkXero-dev** [comment]
Sorry it took so long just got Plasma 6 on Arch and nope ain't working. Also I would argue we don't really 
need it coz look at what I did with stock Plasma clock widget ..  
![image](https://github.com/MarianArlt/kde-plasmoid-betterinlineclock/assets/81101439/559a674b-929b-4236-9e6b-11a97c45e1c3)

Your fork error out
![image](https://github.com/MarianArlt/kde-plasmoid-betterinlineclock/assets/81101439/4f38c7ae-5284-41b3-aa77-24c112302d4e)

**Comment by DuckySoLucky** [comment]
Are there any plans on porting it onto Plasma 6?

**Comment by DarkXero-dev** [comment]
> Are there any plans on porting it onto Plasma 6?

You can do the same with Stock Plasma version no need for Inline IMO..

**Comment by DuckySoLucky** [comment]
are we talking about this one or?

![image](https://github.com/MarianArlt/kde-plasmoid-betterinlineclock/assets/75372052/f08cc84e-3148-4d1c-835d-3f5351dae6da)

**Comment by DarkXero-dev** [comment]
> DuckySoLucky

Yes .. Right Click > Configure > set it up as you wish  .. My settings 
![image](https://github.com/MarianArlt/kde-plasmoid-betterinlineclock/assets/81101439/57d9f125-0e63-41a5-8272-64ac149b75b6)

**Comment by DuckySoLucky** [comment]
ty ily :heart:

**Comment by Svenum** [comment]
> > DuckySoLucky
> 
> Yes .. Right Click > Configure > set it up as you wish .. My settings ![image](https://private-user-images.githubusercontent.com/81101439/310637677-57d9f125-0e63-41a5-8272-64ac149b75b6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk4OTk5MjEsIm5iZiI6MTcwOTg5OTYyMSwicGF0aCI6Ii84MTEwMTQzOS8zMTA2Mzc2NzctNTdkOWYxMjUtMGU2My00MWE1LTgyNzItNjRhYzE0OWI3NWI2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzA4VDEyMDcwMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQyOTgxNTE3OWRhMjJjMWE3ZGFhYzFhN2EwNjdjZTBjNmI1YjYyYjNkOGM0MmJiYzQ5ZWNiYjkxNTgzYjMzMzUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.VmehannaGmBb1vWeSDPIL_Rh6K7bK6emRAJvtXo4cEM)

Thats great man. Thanks

**Comment by aksh1618** [comment]
> > DuckySoLucky
> 
> Yes .. Right Click > Configure > set it up as you wish .. My settings

Awesome, thanks!! If only there were a way to also show seconds in the time part, that would account for all my use cases for this plasmoid!

I also tried setting the date format to `yyyy-MM-dd hh:mm:ss | ` but as expected, the date part is not being updated as regularly as the time part:
![image](https://github.com/MarianArlt/kde-plasmoid-betterinlineclock/assets/26065812/29a1212e-a985-4658-9bb8-fb68a549011a)

**Comment by owl-from-hogvarts** [comment]
This is done. See here https://invent.kde.org/plasma/plasma-workspace/-/merge_requests/4073

**Comment by owl-from-hogvarts** [comment]
The above MR was converted to `feat: add separator`. This feature is considered the most valuable out of all features, introduced by `better-inline-clock`. Other features like vertical and horizontal offsets, custom spacing were discarded by KDE Developers. 

If you really miss any of these features, I can publish the original ported version of `better-inline-clock` as PR here. 

If I miss any other noticeable features of the applet which you would like to see in `digital-clock` shipped with KDE Plasma, tag me here please

**Comment by hydd30** [comment]
> > DuckySoLucky
> 
> Yes .. Right Click > Configure > set it up as you wish .. My settings ![image](https://private-user-images.githubusercontent.com/81101439/310637677-57d9f125-0e63-41a5-8272-64ac149b75b6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTY0NzQ2NTQsIm5iZiI6MTcxNjQ3NDM1NCwicGF0aCI6Ii84MTEwMTQzOS8zMTA2Mzc2NzctNTdkOWYxMjUtMGU2My00MWE1LTgyNzItNjRhYzE0OWI3NWI2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTIzVDE0MjU1NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTk1ZmE4YzAyMTRiNTM3YTEyNWY4Y2RiMDg0NzJmZTA0ZjBlNGRmNWYxYzFiN2IyNTk1ZTY3NDg3ZTQ3NzEwMTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.6We5LcDAjuCENRXm00mJ8nAAYEK-Ktu5Ma4KPGJKcx0)

But I met the problem using this that the time has the bigger font than the date while they are the same in this better-inline-clock. Whether I miss some settings or it's just the feature(bug)?

**Comment by hydd30** [comment]
> > DuckySoLucky
> 
> Yes .. Right Click > Configure > set it up as you wish .. My settings ![image](https://private-user-images.githubusercontent.com/81101439/310637677-57d9f125-0e63-41a5-8272-64ac149b75b6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTY0NzQ2NTQsIm5iZiI6MTcxNjQ3NDM1NCwicGF0aCI6Ii84MTEwMTQzOS8zMTA2Mzc2NzctNTdkOWYxMjUtMGU2My00MWE1LTgyNzItNjRhYzE0OWI3NWI2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTIzVDE0MjU1NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTk1ZmE4YzAyMTRiNTM3YTEyNWY4Y2RiMDg0NzJmZTA0ZjBlNGRmNWYxYzFiN2IyNTk1ZTY3NDg3ZTQ3NzEwMTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.6We5LcDAjuCENRXm00mJ8nAAYEK-Ktu5Ma4KPGJKcx0)

But I met the problem using this that the time has the bigger font than the date while they are the same in this better-inline-clock. Whether I miss some settings or it's just the feature(bug)?
![image](https://github.com/MarianArlt/kde-plasmoid-betterinlineclock/assets/52079469/4cd1ddd7-cb8b-464a-ad27-1472dc4ed65e)

**Comment by micmalti** [comment]
> > Are there any plans on porting it onto Plasma 6?
> 
> You can do the same with Stock Plasma version no need for Inline IMO..

Not exactly. The default clock widget in Plasma supports the inclusion of an event calendar only through KOrganizer, while Better Inline Clock avoids it completely. More importantly, however, is the lack of support to add/edit events directly from the widget. This is the only feature for which I couldn't find an alternative after the upgrade.
